core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadRequest.java

```diff
@@ -74,17 +78,10 @@ private DownloadRequest(Parcel in) {
         feedfileType = in.readInt();
         lastModified = in.readString();
         deleteOnFailure = (in.readByte() > 0);
+        username = nullIfEmpty(in.readString());
+        password = nullIfEmpty(in.readString());
+        mediaEnqueued = (in.readByte() > 0);
         arguments = in.readBundle();
-        if (in.dataAvail() > 0) {
```

Comment:
You changed the order and layout of the parcel. Do you think this could lead to problems with existing parcels from old app versions? Do we need some kind of migration? I think those are only stored temporarily, so this should not be a problem but you know that code better, so I would be happy about a confirmation.

Reply:
I don't think migration is needed, and have rechecked all references to `DownloadRequest` to reconfirm it.

The marshalling/unmarshalling happens only in making a service request. Otherwise, all DownloadRequests objects are stored in-memory in `DownloadService`. Even if there is some OS work that result in the in-memory objects marshalled, they would still be temporary in nature.